services/w3c/w3c-validation.service.js

```diff
@@ -0,0 +1,269 @@
+    return helpers.error('regex')
+  }
+
+  // Return the value unchanged
+  return value
+}
+
+const queryParamSchema = Joi.object({
+  doc: optionalUrl.required(),
+  preset: Joi.string()
+    .allow('')
+    .custom(queryParamDecode(presetRegex), 'preset validation'),
+  parser: Joi.string()
+    .allow('')
+    .custom(queryParamDecode(parserRegex), 'parser validation'),
+}).required()
+
+const documentation = `
+  <style>
+    .box {
+      display: flex;
+      justify-content: space-between;
+    }
+    .note {
+      font-size: smaller;
+      text-align: left;
+    }
+  </style>
+  <p>
+    The W3C validation badge performs validation of the HTML, SVG, MathML, ITS, RDFa Lite, XHTML documents.
+    The badge uses the type property of each message found in the messages from the validation results to determine to be an error or warning.
+    The rules are as follows:
+    <ul class="note">
+      <li>info:  These messages are counted as warnings</li>
+      <li>error:  These messages are counted as errors</li>
+      <li>non-document-error: These messages are counted as errors</li>
+    </ul>
+  </p>
+  <p>
+    This badge relies on the https://validator.nu service to perform the validation. Please refer to https://about.validator.nu/ for the full documentation and Terms of service.
+    For the badge to function it requires the consumer to promote the following parameters:
+    <ul class="note">
+      <li>
+        doc (Required): This is the path for the document to be validated
+      </li>
+      <li>
+        preset (Optional can be left as blank): This is used to determine the schema for the document to be valdiated against.
+        The following are the allowed values
+        <ul>
+          <li>HTML, SVG 1.1, MathML 3.0</li>
+          <li>HTML, SVG 1.1, MathML 3.0, ITS 2.0</li>
+          <li>HTML, SVG 1.1, MathML 3.0, RDFa Lite 1.1</li>
+          <li>HTML 4.01 Strict, URL / XHTML 1.0 Strict, URL</li>
+          <li>HTML 4.01 Transitional, URL / XHTML 1.0 Transitional, URL</li>
+          <li>HTML 4.01 Frameset, URL / XHTML 1.0 Frameset, URL</li>
+          <li>XHTML, SVG 1.1, MathML 3.0</li>
+          <li>XHTML, SVG 1.1, MathML 3.0, RDFa Lite 1.1</li>
+          <li>XHTML 1.0 Strict, URL, Ruby, SVG 1.1, MathML 3.0</li>
+          <li>SVG 1.1, URL, XHTML, MathML 3.0</li>        
+        </ul>
+      </li>    
+      <li>
+        parser (Optional can be left as blank): The parser that is used for validation. This is a passthru value to the service
+        <ul>
+          <li>html <i>(HTML)</i></li>
+          <li>xml <i>(XML; don’t load external entities)</i></li>
+          <li>xmldtd <i>(XML; load external entities)</i></li>
+        </ul>
+      </li>  
+    </ul>
+  </p>
+`
+
+module.exports = class W3cValidation extends BaseJsonService {
+  static get category() {
+    return 'analysis'
+  }
+
+  static get route() {
+    return {
+      base: 'w3c-validation',
+      pattern: 'doc',
```

Comment:
```suggestion
      pattern: ':prefix(default|html|xml|xmldtd)',
```

Reply:
This is the new pattern `pattern: ':parser(default|html|xml|xmldtd)',`